packages/ui-extensions/src/surfaces/point-of-sale/render/components/shared/box.ts

```diff
@@ -0,0 +1,128 @@
+import {SizeKeyword} from './sizes';
+
+export type PaddingKeys = SizeKeyword | 'none';
+
+export interface PaddingProps {
+  /**
+   * Adjust the padding of all edges.
+   *
+   * @default '0'
+   */
+  padding?: PaddingKeys;
+
+  /**
+   * Adjust the block-padding.
+   *
+   * - `large none` means block-start padding is `large`, block-end padding is `none`.
+   *
+   * This overrides the block value of `padding`.
+   *
+   * @default '0'
+   */
+  paddingBlock?: PaddingKeys;
+
+  /**
+   * Adjust the block-start padding.
+   *
+   * This overrides the block-start value of `paddingBlock`.
+   *
+   * @default '0'
+   */
+  paddingBlockStart?: PaddingKeys;
+
+  /**
+   * Adjust the block-end padding.
+   *
+   * This overrides the block-end value of `paddingBlock`.
+   *
+   * @default '0'
+   */
+  paddingBlockEnd?: PaddingKeys;
+
+  /**
+   * Adjust the inline padding.
+   *
+   * - `large none` means inline-start padding is `large`, inline-end padding is `none`.
+   *
+   * This overrides the inline value of `padding`.
+   *
+   * @default '0'
+   */
+  paddingInline?: PaddingKeys;
+
+  /**
+   * Adjust the inline-start padding.
+   *
+   * This overrides the inline-start value of `paddingInline`.
+   *
+   * @default '0'
+   */
+  paddingInlineStart?: PaddingKeys;
+
+  /**
+   * Adjust the inline-end padding.
+
+   * @default '0'
+   */
+  paddingInlineEnd?: PaddingKeys;
+}
+
+export type SizeUnits = `${number}px` | `${number}%` | `0`;
+export type SizeUnitsOrNone = SizeUnits | 'none';
+export type SizeUnitsOrAuto = SizeUnits | 'auto';
+
+export interface SizingProps {
+  /**
+   * Adjust the block size.
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/CSS/block-size
+   *
+   * @default 'auto'
```

Comment:
Maybe a comment explaining that auto will take as much space as it needs to layout the children would be useful?